test/test_foreach.py

```diff
@@ -173,14 +184,71 @@ def test_add_list_error_cases(self, device):
             torch._foreach_add_(tensors1, tensors2)
 
     @dtypes(*torch.testing.get_all_dtypes())
-    def test_add_list_same_size(self, device, dtype):
-        tensors1 = [torch.zeros(10, 10, device=device, dtype=dtype) for _ in range(10)]
-        tensors2 = [torch.ones(10, 10, device=device, dtype=dtype) for _ in range(10)]
+    def test_bin_op_list(self, device, dtype):
+        if dtype == torch.bool:
+            return
```

Comment:
What happens with bool? Should you be checking for runtime error with bool instead of bailing out completely?

Reply:
Fixed